tests/components/control4/test_config_flow.py

```diff
@@ -0,0 +1,199 @@
+"""Test the Control4 config flow."""
+import datetime
+
+from pyControl4.error_handling import Unauthorized
+
+from homeassistant import config_entries, setup
+from homeassistant.components.control4.const import DEFAULT_SCAN_INTERVAL, DOMAIN
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+
+from tests.async_mock import AsyncMock, patch
+from tests.common import MockConfigEntry
+
+
+def _get_mock_c4_account(
+    getAccountControllers={
+        "controllerCommonName": "control4_model_00AA00AA00AA",
+        "href": "https://apis.control4.com/account/v3/rest/accounts/000000",
+        "name": "Name",
+    },
+    getDirectorBearerToken={
+        "token": "token",
+        "token_expiration": datetime.datetime(2020, 7, 15, 13, 50, 15, 26940),
+    },
+):
+    c4_account_mock = AsyncMock()
+    type(c4_account_mock).getAccountControllers = AsyncMock(
```

Comment:
I think that this is wrong. 

```suggestion
    c4_account_mock.getAccountControllers.return_value = getAccountControllers
```

Does this work ?

Reply:
And since you want to mock a constructor, you can patch the return value. Updated above suggestion.

Reply:
```c4_account_mock.getAccountControllers.return_value = getAccountControllers``` ended up working. I'm going to push this change and see if that fixes py37 CI test.

Reply:
I'm still getting `TypeError: object MagicMock can't be used in 'await' expression` on py37, even though I've never imported or used `MagicMock` anywhere. 

I see that `CoroutineMock` should be imported for Python < 3.8
```
if sys.version_info[:2] < (3, 8):
    from asynctest.mock import *  # noqa
    from asynctest.mock import CoroutineMock as AsyncMock  # noqa
else:
    from unittest.mock import *  # noqa
```
and if that were true then there should be no way for `MagicMock `to appear since `CoroutineMock `inherits directly from `Mock `in the `asynctest `package.
```
class CoroutineMock(Mock):
```
This leads me to think that `AsyncMock` is incorrectly being imported directly from `unittest.mock` even on Python 3.7.8, as that inherits from some "magic" things:
```
class AsyncMock(AsyncMockMixin, AsyncMagicMixin, Mock):
```
What do you think?

Reply:
Never mind, I get the same error even when explicitly importing `from asynctest.mock import CoroutineMock as AsyncMock`